### DISCUSSION TITLE
Flattening with Projeciton(ProjectTo)

### DISCUSSION THREAD

**Discussion by crowz4k** [LuckyPennySoftware/AutoMapper#3724] [NA]
According to documentation ([Flattening](https://docs.automapper.org/en/stable/Flattening.html)) AutoMapper can flatten structure. 
```cs
    public class Order
    {
        public int Id { get; set; }
        public Order()
        {
            OrderLineItems = new List<OrderLineItem>();
        }

        public virtual Customer Customer { get; set; }
        public virtual  List<OrderLineItem> OrderLineItems { get; set; }
        
        public decimal GetTotal()
        {
            return OrderLineItems.Sum(li => li.GetTotal());
        }
    }

    public class OrderLineItem
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public Product Product { get; set; }

        public int Quantity { get; set; }

        public decimal GetTotal()
        {
            return Quantity * Product.Price;
        }
    }

    public class Product
    {
        public int Id { get; set; }
        public decimal Price { get; set; }
        public string Name { get; set; }
    }

    public class OrderDto
    {
        public string CustomerName { get; set; }
        public decimal Total { get; set; }
    }
```

```cs
var result = await _context.Orders.AsQueryable()
                .ProjectTo<OrderDto>(_mapper.ConfigurationProvider)
                .ToListAsync();
```

But when I try to do that with SQLite and example from the documentation it does not work. Do Flattening work with ProjectTo?

**Comment by lbargaoanu** [comment]
It does work :) Check [the execution plan](https://docs.automapper.org/en/latest/Understanding-your-mapping.html).

**Reply by crowz4k** [reply]
Do I do something wrong?
Here is my sample repo ->  [sample repo](https://github.com/crowz4k/TestAPI)

**Reply by lbargaoanu** [reply]
Start with the most basic example and work your way up. I'm sure you're missing smth.

**Reply by crowz4k** [reply]
You can't get simpler than this, this is from the documentation. You can see I added miniprofiler as well so I get the exact query that it is produced.

```sql
SELECT "c"."Name", "o"."Id", "o"."CustomerId"
FROM "Orders" AS "o"
LEFT JOIN "Customers" AS "c" ON "o"."CustomerId" = "c"."Id"
```

**Reply by jbogard** [reply]
You're using `ProjectTo` with a calculated method. That only works with some LINQ providers. Remove AutoMapper from the equation, use LINQ directly, and you'll see.

**Reply by crowz4k** [reply]
@jbogard Thank you, I wanted to try to do this with AutoMapper so when I pull back data I just want few columns not all, I thought I can use this with AutoMapper Flattening. 
Actually, the real use-case is more complex (EF + SQLServer), I thought I could use ProjectTo
